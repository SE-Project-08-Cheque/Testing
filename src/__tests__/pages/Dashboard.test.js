// import dependencies
import React from "react";
// import react dom
import ReactDOM from "react-dom";
// import react-testing methods
import { render, cleanup, fireEvent } from "@testing-library/react";
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import Dashboard from "../../pages/Dashboard";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  const user = {
    user_type: "Student",
    user_id: "970590054V",
    username: "Chalindu",
    account_no: "123"
}
  ReactDOM.render(
    <MemoryRouter>
      <Dashboard 
        user_type= "Student"
        user_id= "970590054V"
        username= "Chalindu"
        account_no= "123"
      />
    </MemoryRouter>,
    div
  );
});

it("renders Dashboard page correctly", () => {

  const { getByTestId } = render(
    <MemoryRouter>
      <Dashboard 
        user_type= "Student"
        user_id= "970590054V"
        username= "Chalindu"
        account_no= "123"
      />
    </MemoryRouter>
  );
  expect(getByTestId("user_type")).toHaveTextContent("Student");
  expect(getByTestId("user_id")).toHaveTextContent("970590054V");
  expect(getByTestId("username")).toHaveTextContent("Chalindu");
  expect(getByTestId("account_no")).toHaveTextContent("123");
});


it("matches snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Dashboard user_type= "Student"
        user_id= "970590054V"
        username= "Chalindu"
        account_no= "123" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
