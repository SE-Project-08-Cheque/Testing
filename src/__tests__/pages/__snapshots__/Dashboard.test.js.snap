// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="dashboard"
>
  <div
    className="css-ure0nq"
  >
    <div
      className="css-dbpzvv"
    >
      <div
        className="css-1rul54"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="chakra-text css-kbn7if"
          >
            
          </p>
        </div>
        <div
          className="css-sc5c3h"
        >
          <span
            className="chakra-badge css-ifsvay"
          >
            Online
          </span>
        </div>
        <div
          className="css-16n1mrt"
        >
          <div
            aria-valuemax={100}
            aria-valuemin={0}
            className="css-12om7f"
            data-indeterminate=""
            role="progressbar"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-8atqhb"
      >
        <div
          className="chakra-stack css-zsoya5"
        >
          <a
            href="/home"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="overview"
              type="button"
            >
              Overview
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Account_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="personal_details"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Personal Details
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Contact_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="contact_details"
              type="button"
            >
              Contact Details
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/AllNotifications"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="notification"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Notification
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/ForgotPin"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="forgot_pin"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Fogot Pin
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <hr
            aria-orientation="horizontal"
            className="chakra-divider css-1upb9tn"
            role="separator"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1d9x0xk"
  >
    <div
      className="css-1y2qylg"
    >
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-rtmwg4"
        />
      </div>
      <div
        className="css-1orkoap"
      />
      <div
        className="css-9hhi3x"
      >
        <p
          className="chakra-text css-epm5mk"
          data-testid="user_type"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
                />
                <path
                  d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
                />
              </g>
            </svg>
          </button>
          User Type : 
          Student
           
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-epm5mk"
          data-testid="user_id"
        >
          <button
            aria-label="Call"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 14 14"
            >
              <path
                d="M2.20731,0.0127209 C2.1105,-0.0066419 1.99432,-0.00664663 1.91687,0.032079 C0.871279,0.438698 0.212942,1.92964 0.0580392,2.95587 C-0.426031,6.28627 2.20731,9.17133 4.62766,11.0689 C6.77694,12.7534 10.9012,15.5223 13.3409,12.8503 C13.6507,12.5211 14.0186,12.037 13.9993,11.553 C13.9412,10.7397 13.186,10.1588 12.6051,9.71349 C12.1598,9.38432 11.2304,8.47427 10.6495,8.49363 C10.1267,8.51299 9.79754,9.05515 9.46837,9.38432 L8.88748,9.96521 C8.79067,10.062 7.55145,9.24878 7.41591,9.15197 C6.91248,8.8228 6.4284,8.45491 6.00242,8.04829 C5.57644,7.64167 5.18919,7.19632 4.86002,6.73161 C4.7632,6.59607 3.96933,5.41495 4.04678,5.31813 C4.04678,5.31813 4.72448,4.58234 4.91811,4.2919 C5.32473,3.67229 5.63453,3.18822 5.16982,2.45243 C4.99556,2.18135 4.78257,1.96836 4.55021,1.73601 C4.14359,1.34875 3.73698,0.942131 3.27227,0.612963 C3.02055,0.419335 2.59457,0.0708094 2.20731,0.0127209 Z"
                fill="currentColor"
              />
            </svg>
          </button>
          User ID : 
          970590054V
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-qxtfng"
          data-testid="username"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
                fill="currentColor"
              />
            </svg>
          </button>
          User Name : 
          Chalindu
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-qxtfng"
          data-testid="account_no"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
                />
                <path
                  d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
                />
              </g>
            </svg>
          </button>
          User Account : 
          123
        </p>
      </div>
    </div>
  </div>
</div>
`;
